pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -861,83 +883,114 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		networks[network.Name] = network.DeepCopy()
 	}
 
+	sriovPciAddresses := getSriovPciAddresses()
+
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
 		net, isExist := networks[iface.Name]
 		if !isExist {
 			return fmt.Errorf("failed to find network %s", iface.Name)
 		}
 
-		ifaceType := getInterfaceType(&iface)
-		domainIface := Interface{
-			Model: &Model{
-				Type: ifaceType,
-			},
-			Alias: &Alias{
-				Name: iface.Name,
-			},
-		}
-
-		// if UseEmulation unset and at least one NIC model is virtio,
-		// /dev/vhost-net must be present as we should have asked for it.
-		if ifaceType == "virtio" && virtioNetProhibited {
-			return fmt.Errorf("In-kernel virtio-net device emulation '/dev/vhost-net' not present")
-		} else if ifaceType == "virtio" && virtioNetMQRequested {
-			domainIface.Driver = &InterfaceDriver{Name: "vhost", Queues: numQueues}
-		}
-
-		// Add a pciAddress if specifed
-		if iface.PciAddress != "" {
-			addr, err := decoratePciAddressField(iface.PciAddress)
+		if iface.Sriov != nil {
+			var pciAddr string
+			pciAddr, sriovPciAddresses, err = popSriovPciAddress(sriovPciAddresses)
 			if err != nil {
-				return fmt.Errorf("failed to configure interface %s: %v", iface.Name, err)
+				return err
 			}
-			domainIface.Address = addr
-		}
 
-		if iface.Bridge != nil {
-			// TODO:(ihar) consider abstracting interface type conversion /
-			// detection into drivers
-			domainIface.Type = "bridge"
-			if value, ok := cniNetworks[iface.Name]; ok {
-				prefix := ""
-				// no error check, we assume that CNI type was set correctly
-				if net.Multus != nil {
-					prefix = "net"
-				} else if net.Genie != nil {
-					prefix = "eth"
-				}
-				domainIface.Source = InterfaceSource{
-					Bridge: fmt.Sprintf("k6t-%s%d", prefix, value),
-				}
-			} else {
-				domainIface.Source = InterfaceSource{
-					Bridge: DefaultBridgeName,
-				}
+			dbsfFields, err := util.ParsePciAddress(pciAddr)
+			if err != nil {
+				return err
 			}
 
-			if iface.BootOrder != nil {
-				domainIface.BootOrder = &BootOrder{Order: *iface.BootOrder}
+			hostDev := HostDevice{
```

Comment:
Can you add  the bootOrder also ?
https://libvirt.org/formatdomain.html#elementsHostDevSubsys

Reply:
Thanks for pointing out I missed it, added it in the latest upload.